ClientApp/src/app/sprint-dropdown/sprint-dropdown.component.ts

```diff
@@ -0,0 +1,31 @@
+import { Component, OnInit, Input } from '@angular/core';
+
+import { ServerApi } from '../server-api/server-api.service';
+import { SprintData } from '../../types/sprint';
+import { TaskData } from '../../types/task';
+
+@Component({
+  selector: 'sprint-dropdown',
+  templateUrl: './sprint-dropdown.component.html',
+  styleUrls: ['./sprint-dropdown.component.css' ],
+})
+export class SprintDropdownComponent implements OnInit {
+  @Input() id!: number;
+  
+  data!: SprintData;
+
+  tasks!: TaskData[];
+  
+  constructor(private serverApi: ServerApi) { }
+
+  ngOnInit(): void {
+    this.tasks = [];
+    this.loadSprintData();
+  }
+
+  loadSprintData(): void {
+    this.data = this.serverApi.getSprintData(this.id);
+    this.tasks.push(this.serverApi.getTaskData(10000));
```

Comment:
So if I'm understanding correctly, in the actual implementation, we'd have to get the IDs for tasks related to this sprint and then either query them one by one? I wonder if we should have a `getTaskDatafromSprintID` API that returns an array

Reply:
Yeah, I've thought about that too. If we decide to query a sprint's tasks one by one, it might increase latency between frontend and backend by quite a bit, especially for the backlog which could potentially have hundreds or thousands of tasks. It might be better to have a `getAllTasks(sprintId)` so they can be lumped into a single http request. Would that be easier or more difficult on the backend <--> database side, do you think?

Reply:
I think it would be about equal in difficulty, the HTTP protocol should handle large responses without us having to worry about it on the application layer.